### ISSUE TITLE
Build broken

### ISSUE THREAD

**Issue by lucyb** [opensafely-core/interactive-templates#363]
The [build is broken](https://github.com/opensafely-core/interactive-templates/actions?query=branch%3Amain) on main and has been for a little while.

It's failing on the tag new release step, although it's unclear why

```
To https://github.com/opensafely-core/interactive-templates
 ! [rejected]        v2023.09.12.100407 -> v2023.09.12.100407 (already exists)
error: failed to push some refs to 'https://github.com/opensafely-core/interactive-templates'
hint: Updates were rejected because the tag already exists in the remote.
Error: Process completed with exit code 1.
```

**Comment by lucyb** [comment]
I am uncertain whether this is related or not, but it's worth being aware of just in case.
https://github.com/opensafely-core/interactive-templates/issues/118

**Comment by StevenMaude** [comment]
- [ ] https://github.com/opensafely-core/interactive-templates/pull/365

I could also bump the version as part of this issue, which I can do afterwards.

**Comment by lucyb** [comment]
Thanks for digging into this. I understand your point about it being a bit quirky. Would using conventional commits allow us to easily tag a new release in the same way we're doing now? Job Server should be picking up any new tags and raising a PR for us, and ideally we'd want that to stay the same.

Also, Louis is one of the main committers to this repo. How would he feel about using conventional commits? I suspect he'd be fine, but we'd have to tell him about the change.

**Comment by StevenMaude** [comment]
@lucyb: Ah, looking around I didn't have the full context here.

The [`pipeline`](https://github.com/opensafely-core/pipeline/) repository works in the same way with a `version` file.

While there's an [job-server ADR for having this interactive-templates](https://github.com/opensafely-core/job-server/blob/5670e3b8be323683e21f53110c8de44ea5c3789e/docs/adr/0013-interactive-templates-repo.md), it's not clear why the tags are date-based, and opposed to a sequential version, and I can't easily find out why. Maybe recording the date is useful in some way?

I'd add that the "conventional commits" system we have elsewhere is also quirky. Just that I'm more familiar with it since I'd seen it in the ehrQL repository and elsewhere.
 
So, rather than get embroiled in that, I'll propose that we:

1. fix the build as stated above, by merging the PR
1. update the version file to trigger a new release
1. verify that all is well with the fix
1. document the version bump process in the developer notes or README

but leave the versioning process alone for now.

If the specifics of handling updates are annoying enough, then we can open a new issue for that, but it's probably fine as these are probably infrequent.

**Comment by Jongmassey** [comment]
I just started looking at this because my actions failed for this reason. I think I agree with Steve's suggested solution

**Comment by StevenMaude** [comment]
Actually, I'll wait until #364 is merged as it looks nearly done, and then release.

There's also #368, but that might take a little longer to unpick.

**Comment by StevenMaude** [comment]
Reopening as the automatic PR to job-server failed. I'll take a look.

**Comment by StevenMaude** [comment]
opensafely-core/job-server#4437 is related to this.

**Comment by StevenMaude** [comment]
The build's fine; it's the creating PR workflow that's an issue now.

**Comment by StevenMaude** [comment]
Proposal to move the PR workflow to job-server: https://github.com/opensafely-core/job-server/pull/4442 — it means the PRs are not created on immediate release of a new version here, but does remove the use of a PAT, and is how things work for the ehrQL repository.

**Comment by StevenMaude** [comment]
Once opensafely-core/job-server#4456 is merged, that part is complete. The workflow there works.

What's left:

* update the developer notes here
* remove the currently non-functioning workflow here
